### PR TITLE
Add /debug/events and /debug/requests

### PR DESCRIPTION
**Pull request by mhilton**
Add the golang.org/x/net/trace endpoints to the standard debugstatus handler.

(Review request: http://reviews.vapour.ws/r/4230/)

### CODE REVIEW

File: debugstatus/handler.go [hunk 1]
```diff
@@ -41,6 +42,15 @@ type Handler struct {
 	// of the endpoints under /debug/pprof - the
 	// error returned will be ErrNoPprofConfigured.
 	CheckPprofAllowed func(req *http.Request) error
+
+	// CheckTraceAllowed will be used to check whether the given
+	// trace request should be allowed. It should return an error if
+	// not, which will not be masked. If this is nil, no access will
+	// be allowed to either /debug/events or /debug/requests - the
+	// error returned will be ErrNoTraceConfigured. If access is
+	// allowed the sensative boolean is passed on the the trace
```
Comment (rogpeppe) [inline debugstatus/handler.go]: s/sensative/sensitive/

The trace implementation seems to ignore the sensitive parameter
though - I wonder how it's meant to work.

File: debugstatus/handler.go [hunk 2]
```diff
@@ -41,6 +42,15 @@ type Handler struct {
...
+	// handles.
+	CheckTraceAllowed func(req *http.Request) (err error, sensitive bool)
```
Comment (rogpeppe) [inline debugstatus/handler.go]: putting the error last would be more conventional.

File: debugstatus/handler.go (hunk 1 above)
Comment (rogpeppe) [inline debugstatus/handler.go]: Perhaps:

// If access is allowed, the sensitive value specifies whether
// sensitive trace events will be shown.

?

File: debugstatus/handler.go [hunk 3]
```diff
@@ -116,3 +126,46 @@ func (h *Handler) checkPprofAllowed(req *http.Request) error {
...
+// DebugEventsRequest describes the /debug/events endpoint.
+type DebugEventsRequest struct {
+	httprequest.Route `httprequest:"GET /debug/events"`
+}
+
+// DebugEvents serves the /debug/events endpoint.
+func (h *Handler) DebugEvents(p httprequest.Params, r *DebugEventsRequest) error {
+	err, sensitive := h.checkTraceAllowed(p.Request)
+	if err != nil {
+		return err
```
Comment (rogpeppe) [inline debugstatus/handler.go]: return errgo.Mask(err, errgo.Any)

(and below)
?

**Comment by rogpeppe** [comment]
Nice stuff. LGTM with a few minor thoughts above, thanks.

**Comment by urosj** [comment]
:+1: Ty!

**Comment by mhilton** [comment]
$$merge$$

**Pull request merged**